### PR TITLE
Convert OperatorUI Button to Typescript

### PR DESCRIPTION
**Pull request by rupurt**
- Replace runtime checks with type annotations

### CODE REVIEW

File: operator_ui/src/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  | undefined
+
+export type ButtonVariant =
+  | MuiButtonVariant
+  | 'primary'
+  | 'secondary'
+  | 'danger'
+  | 'default'
+
+const muiProps = (variant: ButtonVariant, classes: any): MuiButtonProps => {
```
Comment (rupurt) [inline operator_ui/src/components/Button.tsx]: @smartcontractkit/core-devs any idea how to extract this type annotation?

https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/styles/withStyles.d.ts#L40

Comment (NavyAdmiral) [reply operator_ui/src/components/Button.tsx]: @rupurt hmm, are you asking how ClassNameMap's typed? https://github.com/mui-org/material-ui/blob/master/packages/material-ui-styles/src/withStyles/withStyles.d.ts#L68 
If MUI doesn't export the needed annotations, we can duplicate definitions from their codebase

File: operator_ui/src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import MuiButton, {
+  ButtonProps as MuiButtonProps
+} from '@material-ui/core/Button'
+import { TouchRippleProps } from '@material-ui/core/ButtonBase/TouchRipple'
+import classNames from 'classnames'
+
+const styles = (theme: Theme) =>
+  createStyles({
+    default: {
+      borderColor: '#BECAD6',
```
[the remark this answers is not in the document]
Comment (rupurt) [reply operator_ui/src/components/Button.tsx]: @NavyAdmiral it should be bulled from the theme. However adding custom theme attributes requires additional changes to the type annotations which I'm trying to leave for [chores/operator-ui-strict-typescript](https://github.com/smartcontractkit/chainlink/tree/chores/operator-ui-strict-typescript)

File: operator_ui/src/components/Button.tsx (hunk 1 above)
Comment (rupurt) [reply operator_ui/src/components/Button.tsx]: @NavyAdmiral I'm wondering if there's any way around duplicating it. Like is there a special escape hatch. We're on `v3.9.2` which is different to the latest `v4.2.1` and I'd prefer not to try and keep them in sync. 

What's your thoughts on just leaving it as `any` with a comment if there is no way to pull it from the library?

Comment (NavyAdmiral) [reply operator_ui/src/components/Button.tsx]: Historically, if these types didn't change much across MUI versions, then we could use `patch-package` and export the needed type manually. Personally, I'd be good with just leaving it as `any`

Comment (rupurt) [reply operator_ui/src/components/Button.tsx]: 👍 `patch-package` seems pretty cool. Normally I'd just maintain a fork until it's been merged upstream but this is another good option.

File: operator_ui/src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      '&:hover': {
+        backgroundColor: theme.palette.common.white,
+        borderColor: '#BECAD6',
+        boxShadow:
+          '0 2px 4px 0 rgba(0,123,255,0.06), 0 2px 2px 0 rgba(0,0,0,0.06)'
```
[the remark this answers is not in the document]
Comment (rupurt) [reply operator_ui/src/components/Button.tsx]: The shadow on this button is custom. It's slightly different from the graded box shadows that get referenced by index https://material-ui.com/system/shadows#shadows

[the remark this answers is not in the document]
Comment (rupurt) [reply operator_ui/src/components/Button.tsx]: @thoughtmerchant we added this as a custom style. What's your thoughts on getting this to use the shadow index levels?

File: operator_ui/src/components/Button.tsx (hunk 1 above)
Comment (NavyAdmiral) [reply operator_ui/src/components/Button.tsx]: @rupurt did you make up your mind about this? I'm a passionate user of `patch-package`, but this usecase might be a bit contrived.

Comment (rupurt) [reply operator_ui/src/components/Button.tsx]: @NavyAdmiral going with `any`

**Pull request merged**